Updating Tensorflow losses to version 2.x

okbalefthanded opened the pull request:
This PR updates the Tensorflow implemetation of the losses to newer TF version 2.x. Tested in colab with TF 2.8.
Disclaimer: the test files are not updated yet.

okbalefthanded committed:
- <Okba Bekhelifi fork>
- Updated the TensorFlow implementation to TF version >=  2.0
 On branch master
 Your branch is up to date with 'origin/master'.

 Changes to be committed:
	modified:   README.rst
	modified:   examples/linear_classification_tensorflow.py
	modified:   examples/linear_regression_tensorflow.py
	modified:   fyl_tensorflow.py

mblondel commented:
Thanks a lot for the contribution! I will wait for the tests before merging this.

okbalefthanded committed:
- Updated Tensorflow tests On branch master
 Your branch is up to date with 'origin/master'.

 Changes to be committed:
	modified:   tests/test_fyl_tensorflow.py

okbalefthanded commented:
Sorry for the late update, I've tweaked the old tests with v1.compat to keep backward compatibility. The tests pass well.

mblondel commented:
Thank you! Merged!
